app/controllers/versions_controller.rb

```diff
@@ -0,0 +1,32 @@
+class VersionsController < ApplicationController
+  helper Helpers::Geminabox
+  helper_method :load_gems
+
+  def show
+    # Taken from https://github.com/geminabox/geminabox/blob/0.12.4/lib/geminabox/server.rb#L108-L109
+    gems     = Hash[server_instance.send(:load_gems).by_name]
+    versions = gems[params['gemname']]
+    @version = versions && versions.find { |v| v.number = params['version'] }
+
+    unless @version
+      render nothing: true, status: :not_found
+      return
+    end
+
+    @spec = server_instance.send(:spec_for, @version.name, @version.number)
+  end
+
+  private
+
+  # Allows us to use the instance methods defined in Geminabox::Server.
+  #
+  # We use .new! as defined by Sinatra so we get a class instance and not a
+  # Sinatra::Wrapper instance.
+  def server_instance
+    @server_instance ||= Geminabox::Server.new!
+  end
+
+  def load_gems
```

Comment:
its a `helper_method`. Its used in the helper method `partial_collection` that is used in the view.